tests/src/Api/FacetApiTest.php

```diff
@@ -0,0 +1,113 @@
+ */
+class FacetApiTest extends TestCase
+{
+    use ClientTestTrait;
+    use FacetTestGeneratorTrait;
+    use InspectTestRequestTrait;
+
+    /**
+     * @covers ::setFacetSort
+     */
+    public function testSetSortInvalidParameter(): void
+    {
+        $object = new FacetApi();
+        $class = new \ReflectionClass($object);
+        $method = $class->getMethod('setFacetSort');
+        $exception = new EuropaSearchApiInvalidParameterValueException("::setFacetSort() received invalid argument 'Invalid argument', must be one of 'DATE', 'REVERSE_DATE', 'ALPHABETICAL', 'REVERSE_ALPHABETICAL', 'DOCUMENT_COUNT', 'REVERSE_DOCUMENT_COUNT', 'NUMBER_DECREASING', 'NUMBER_INCREASING'.");
+        $this->expectExceptionObject($exception);
+        $method->invokeArgs($object, ['Invalid argument']);
+    }
+
+    /**
+     * @covers ::getFacets
+     * @dataProvider providerTestGetFacets
+     *
+     * @param array $clientConfig
+     * @param array $responses
+     * @param mixed $expectedResult
+     */
+    public function testGetFacets(array $clientConfig, array $responses, $expectedResult): void
+    {
+        $actualResult = $this->getTestingClient($clientConfig, $responses, [$this, 'inspectRequest'])
+            ->getFacets(
+                'whatever',
+                ['en', 'de'],
+                'en',
+                ['term' => ['DMAKE_ES_EVENT_TYPE_NAME' => 'ADOPTION_DISTRIBUTE']],
+                'ALPHABETICAL',
+                '21edswq223rews'
+            );
+        $this->assertEquals($expectedResult, $actualResult);
+    }
+
+    /**
+     * @param RequestInterface $request
+     */
+    public function inspectRequest(RequestInterface $request): void
+    {
+        $this->assertEquals('http://example.com/facet?apiKey=foo&database=qux&text=whatever&sessionToken=21edswq223rews&sort=ALPHABETICAL', $request->getUri());
+        $this->inspectBoundary($request);
+        $parts = $this->getMultiParts($request);
+        $this->assertCount(3, $parts);
+        $this->inspectPart($parts[0], 'application/json', 'languages', 11, '["en","de"]');
+        $this->inspectPart($parts[1], 'application/json', 'query', 59, '{"term":{"DMAKE_ES_EVENT_TYPE_NAME":"ADOPTION_DISTRIBUTE"}}');
+        $this->inspectPart($parts[2], 'application/json', 'displayLanguage', 4, '"en"');
+    }
+
+    /**
+     * @return array
+     * @see self::testGetFacets
+     */
+    public function providerTestGetFacets(): array
+    {
+        [$facetsAsArray, $facetsAsObject] = $this->generateTestingFacetItems(5);
+        $facetValues = [];
+        for ($i = 0; $i < 5; $i++) {
+            $facetValues[] = [
+                'apiVersion' => str_shuffle(md5(serialize($facetValues))),
+                'count' => rand(30, 3000),
+                'rawValue' => str_shuffle(md5(serialize($facetValues))),
+                'value' => str_shuffle(md5(serialize($facetValues))),
+            ];
+        }
```

Comment:
What is this for? I guess a leftover.

Reply:
Will look into...

Reply:
It comes from this https://github.com/openeuropa/europa-search-client/commit/cb1b513c2091dec1b15fa120583829039f46102a

it can be removed.

Reply:
removed